### PR TITLE
Pinned schema for kryo compatibility

### PR DESCRIPTION
**Pull request by marcinwadon**

### CODE REVIEW

File: modules/shared/src/main/scala/org/tessellation/schema/GlobalSnapshotInfo.scala [hunk 1]
```diff
@@ -58,6 +58,81 @@ object GlobalSnapshotStateProof {
...
+      balances,
+      lastCurrencySnapshots.map {
+        case (k, v) =>
+          (
+            k,
+            v.map {
+              case (Signed(inc, proofs), info) => (Signed(inc.toCurrencyIncrementalSnapshot, proofs), info)
+            }
+          )
+      },
```
Comment (sdavidp) [inline modules/shared/src/main/scala/org/tessellation/schema/GlobalSnapshotInfo.scala:84]: What are your thoughts on this ?
```
      gs.lastCurrencySnapshots.view.mapValues(v =>
        v.map {
          case (Signed(inc, proofs), info) =>
            (Signed(CurrencyIncrementalSnapshotV1.fromCurrencyIncrementalSnapshot(inc), proofs), info)
        }
      ).to(gs.lastCurrencySnapshots.sortedMapFactory),
```

Comment (sdavidp) [reply modules/shared/src/main/scala/org/tessellation/schema/GlobalSnapshotInfo.scala:84]: (I pasted that to the wrong place, .... but similar to code later on.

Comment (marcinwadon) [reply modules/shared/src/main/scala/org/tessellation/schema/GlobalSnapshotInfo.scala:84]: @sdavidp what is `sortedMapFactory`?

Comment (marcinwadon) [reply modules/shared/src/main/scala/org/tessellation/schema/GlobalSnapshotInfo.scala:84]: I see. I've already tried `view.mapValues` but didn't know the trick with sorted map factory. thanks!

**Review by kpudlik — APPROVED** [review]

**Pull request merged**